res/sample-mqtt-msg-bus/configuration.toml

```diff
@@ -0,0 +1,158 @@
+    [Writable.Pipeline.Functions.EncryptWithAES]
+      [Writable.Pipeline.Functions.EncryptWithAES.Parameters]
+        Key = "aquqweoruqwpeoruqwpoeruqwpoierupqoweiurpoqwiuerpqowieurqpowieurpoqiweuroipwqure"
+        InitVector = "123456789012345678901234567890"
+    [Writable.Pipeline.Functions.SetOutputData]
+    [Writable.Pipeline.Functions.MarkAsPushed]
+    [Writable.Pipeline.Functions.PushToCore]
+      [Writable.Pipeline.Functions.PushToCore.Parameters]
+        DeviceName = ""
+        ReadingName = ""
+    [Writable.Pipeline.Functions.HTTPPost]
+      [Writable.Pipeline.Functions.HTTPPost.Parameters]
+        url = "http://"
+        mimeType = "" #OPTIONAL - default application/json
+        persistOnError = "false"
+    [Writable.Pipeline.Functions.HTTPPostJSON]
+      [Writable.Pipeline.Functions.HTTPPostJSON.Parameters]
+        url = "http://"
+        persistOnError = "false"
+    [Writable.Pipeline.Functions.HTTPPostXML]
+      [Writable.Pipeline.Functions.HTTPPostXML.Parameters]
+        url = "http://"
+        persistOnError = "false"
+    [Writable.Pipeline.Functions.MQTTSend]
+      [Writable.Pipeline.Functions.MQTTSend.Parameters]
+        qos="0"
+        key=""
+        autoreconnect="false"
+        retain="false"
+        cert=""
+        persistOnError = "false"
+      [Writable.Pipeline.Functions.MQTTSend.Addressable]
+        Address=   "localhost"
+        Port=      1883
+        Protocol=  "tcp"
+        Publisher= "MyApp"
+        User=      ""
+        Password=  ""
+        Topic=     "sampleTopic"
+
+[Service]
+BootTimeout = "30s"
+ClientMonitor = "15s"
+CheckInterval = "10s"
+Host = "localhost"
+Port = 48095
+Protocol = "http"
+ReadMaxLimit = 100
+StartupMsg = "Sample Configurable Application Service (using MQTT Message Bus) Started"
+Timeout = "5s"
+
+[Registry]
+Host = "localhost"
+Port = 8500
+Type = "consul"
+
+# Database is require when Store and Forward is enabled
+[Database]
+Type = "mongodb"
+Host = "localhost"
+Port = 27017
+Timeout = "30s"
+Username = "appservice"
+Password = "password"
+
+# SecretStore is required when Store and Forward is enabled and running with security
+# so Databse credentails can be pulled from Vault.
+[SecretStore]
+  Host = 'localhost'
+  Port = 8200
+  Path = '/v1/secret/edgex/application-service/'
+  Protocol = 'https'
+
+  [SecretStore.Authentication]
+  AuthType = 'X-Vault-Token'
+  AuthToken = 'edgex'
+
+[Clients]
+  [Clients.CoreData]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 48080
+
+  [Clients.Logging]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 48061
+
+[Binding]
+Type="messagebus"
+SubscribeTopic="events"
+PublishTopic="example"
+
+[MessageBus]
+Type = "mqtt"
+    [MessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.PublishHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.Optional]
```

Comment:
does the go-mod-messaging library handle "Secrets" of these values from the vault?

Reply:
No, we will need to address this during boot strapping.